### PR TITLE
ngFormBuilder translation

### PR DESCRIPTION
**Pull request by AbdallaMohamed**

### CODE REVIEW

File: src/factories/formioTranslate.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  var $translate, gettextCatalog;
+  if($injector.has('$translate')) {
+    $translate = $injector.get('$translate');
+  }
+  if($injector.has('gettextCatalog')) {
+    gettextCatalog = $injector.get('gettextCatalog');
+  }
+  return function (input) {
+    if($translate) {
+      return $translate(input).promise;
```
Comment (AbdallaMohamed) [inline src/factories/formioTranslate.js]: Still need to figure out how $translate's promises work.

**Comment by travist** [comment]
This looks good. We will need to so some testing on our end. I am rolling out a low-risk build right now and will include these PR's in the next build.

Thanks you for your contribution.

**Comment by randallknutson** [comment]
This looks really good. The only thing is we should use the formioTranslate from ngFormio (https://github.com/formio/ngFormio/blob/163aeaa8af2d61a4b260d9ae4fa7184b4a56e5fa/src/formio.js#L63).

ngFormio is already a dependency so it should already be available. No need to recreate it.

**Comment by travist** [comment]
@AbdallaMohamed I want to pull in this pull request, but see you did not respond to @randallknutson comments. Do you think you can modify the formioTranslate filter in the renderer to include this functionality?  If so, then I will move this into the platform.

**Comment by AbdallaMohamed** [comment]
@travist I just pushed a commit and did the things @randallknutson pointed out in his comment. I don't get what you mean by modifying the formioTranslate filter in the renderer though? Could you explain what you exactly need me to do? I'll be happy to make a PR there as well!

**Comment by travist** [comment]
@AbdallaMohamed what he is saying is that in our Renderer @ https://github.com/formio/ngFormio we already define a filter called ```formioTranslate``` that includes our old way of doing it @ https://github.com/formio/ngFormio/blob/163aeaa8af2d61a4b260d9ae4fa7184b4a56e5fa/src/formio.js#L63. It seems you are also creating a filter in this pull request with the same name, so I think they will collide with each other if we pull in this pull request since this repo depends on the Renderer library. 

What we propose is to move your filter changes into the Renderer @ https://github.com/formio/ngFormio and then this library will just pull in those changes since it depends on the renderer. This will also keep the collision from occurring.

Thanks for your contribution.

**Comment by travist** [comment]
Basically, we would like for you to make the changes you made here inside this file so that it works with either translation strategies. https://github.com/formio/ngFormio/blob/163aeaa8af2d61a4b260d9ae4fa7184b4a56e5fa/src/filters/translate.js

**Comment by AbdallaMohamed** [comment]
@travist I made a pull request in the renderer. https://github.com/formio/ngFormio/pull/316
Let me know if that coding style makes sense, or if I should code that in a different way.

**Comment by andreiho** [comment]
@AbdallaMohamed You should remove `src/filters/formioTranslator.js` as well.

**Comment by randallknutson** [comment]
Looks perfect. Thanks for the changes!

**Pull request merged**